pkgs/applications/version-management/helix-cli/default.nix

```diff
@@ -0,0 +1,37 @@
+{ stdenv, fetchurl, patchelf }:
+
+stdenv.mkDerivation rec {
+  name = "helix-cli-${version}";
+  version = "2018.1.1705517";
+
+  src = fetchurl {
+    url = "https://cdist2.perforce.com/perforce/r18.1/bin.linux26x86_64/p4";
+    sha256 = "770c38fbd3726bd11abfd450900d492b41b5bd6049703a063fca20d1a70ecdc6";
+  };
+
+  buildCommand = ''
+    mkdir -p $out/bin/
+    cp $src $out/bin/p4
+    chmod +x $out/bin/p4
+
+    chmod +w $out/bin/p4
+
+    patchelf --interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
```

Comment:
It would be better to do this at `installPhase` (`chmod +w` would then be useless). Note you can use the attribute `dontBuild` to skip the bulid phase if needed.
Also, `install -D -m755 p4 $out/bin/p4` does a lot of job for you :/

Reply:
@nathyong Have you tried actually building this? patchelf doesn't have a `--interpreter` option, you probably meant `--set-interpreter`?

Reply:
@lheckemann I've run `nix-build -A helix-cli` from my copy of `nixpkgs` which succeeds, for some reason. The binary also works too. But I'll change it to `--set-interpreter`.

@nlewo yes, I'd forgotten about the `install` command. I've tried running `patchelf` directly on `$src` during `installPhase` but it still reports "open: Permission denied". Will incorporate your suggestions though, thanks.